### PR TITLE
Unlock folders paper key desktop 584

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

This adds 3 things

#### Multi line input for paper keys, we had a hacked version of this, but:
* it was really hard to style (the content inside overflowed the bounds)
* The text grew downward

## Preview
@keybase/design 

screen shot 2016-03-21 at 6 57 57 pm

## TODO (possibly in a future PR)

The waiting state looks weird on the button, the spinner is too big?

### CODE REVIEW

File: shared/actions/unlock-folders.js [hunk 1]
```diff
@@ -38,10 +39,23 @@ export function toPaperKeyInput (): ToPaperKeyInput {
...
+function waiting (currentlyWaiting: boolean): Waiting {
+  return {type: Constants.waiting, payload: currentlyWaiting}
+}
+
+export function checkPaperKey (paperKey: HiddenString): TypedAsyncAction<CheckPaperKey | Waiting> {
   return dispatch => {
     // TODO figure out what service request to ask for
-    dispatch({type: Constants.checkPaperKey, payload: {success: true}})
+
+    dispatch(waiting(true))
```
Comment (chrisnojima) [inline shared/actions/unlock-folders.js]: i'd just write a comment to use the waiting functionality of the engine and not keep this

**Comment by chrisnojima** [comment]
Is it better to just have <Input multiline> delegate to this component and not expose this directly?

**Comment by chrisnojima** [comment]
:+1: % comment, but would like to discuss

**Comment by MarcoPolo** [comment]
Delegate what to this component?

**Comment by chrisnojima** [comment]
<Input multiline> -> <MultiLineInput>

**Pull request merged**